### ISSUE TITLE
Searchresults getPage() not working "The search request cannot be empty." - and other issues after upgrade from v2 to v3

### ISSUE THREAD

**Issue by Saintenr** [pnp/pnpjs#2326]
### Category
- [x] Bug
- [ ] Question
- [ ] Enhancement
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [@pnp/sp - 3.4.1]

Please specify what version(s) of SharePoint you are targeting: [    SP Online     ]

### Observed Behavior
After upgrading from v2.4 to v3.4, some functions that worked before no longer work.

E.g: I set a search against sharepoint to list files. e.g. 10 per page. I store the initial SearchResults in the react context. when the user later wants to switch to the next page, ctx.searchResults.getPage(2,10) is called. 
In v2.4 this worked without problems. In v3.4 I get the error "The search request cannot be empty."

Has anything changed in the proxy that it doesn't work anymore? does the connection expire or something like that?  Can SearchResults no longer be saved?

If I want to retrieve the 2nd page directly after the initial search request, it works fine.

possibly a similar problem:
I also have the problem that my cache function does not work anymore. I have written in v2.4 with the package localforage the SearchResults as value in an IndexedDB, so that when reloading the page the files as well as filters can be loaded directly. The SearchResults I have then with new SearchResults(obj._raw, obj._url, obj._query, obj._raw, obj._primary); recovered and thus also functions like getPage(). This can't be executed anymore because localForage can't clone the variable. JSON.stringify doesn't make any difference either. I think _url was also renamed to _search? 

Thanks for some answers. I may just have to go back to v2. But since I'm not sure how long this will run productively, I thought to switch to v3 before the project release.

VG Saintenr

**Comment by juliemturner** [comment]
Unfortunately, I would really need more information to begin to help you. If you would share a code repo that replicates your issues I can try and assist, otherwise there just isn't enough context here to even begin to guess what your issues are.

**Comment by Saintenr** [comment]
my project is to large to share this. But now i got another issue. i simply run a normal search query ant gut an unknown error...

I can share my code:
![setup sp](https://user-images.githubusercontent.com/33520976/174745359-33ae6ad3-64c1-4379-ad6f-02f163fb013c.png)
![button function](https://user-images.githubusercontent.com/33520976/174745388-432ba722-fa1c-470f-9d7f-8c5e1402bd73.png)
![error1](https://user-images.githubusercontent.com/33520976/174745830-b6b6058b-f78f-499a-a331-c4b51d34bd8e.png)
![request1](https://user-images.githubusercontent.com/33520976/174745842-52c8a2c9-b1b1-432d-b6ab-747a45bf29a3.png)

**Comment by Saintenr** [comment]
> my project is to large to share this. But now i got another issue. i simply run a normal search query ant gut an unknown error...
> 
> I can share my code: ![setup sp](https://user-images.githubusercontent.com/33520976/174745359-33ae6ad3-64c1-4379-ad6f-02f163fb013c.png) ![button function](https://user-images.githubusercontent.com/33520976/174745388-432ba722-fa1c-470f-9d7f-8c5e1402bd73.png) ![error1](https://user-images.githubusercontent.com/33520976/174745830-b6b6058b-f78f-499a-a331-c4b51d34bd8e.png) ![request1](https://user-images.githubusercontent.com/33520976/174745842-52c8a2c9-b1b1-432d-b6ab-747a45bf29a3.png)

Okay seems that microsoft has massiv issues with search on some tenants. i will update after they fixed this

**Comment by juliemturner** [comment]
Based on the little bit of code you shared I suspect that recreating the context every time a button is clicked is going to lead to issues, you might try reviewing this documentation on a [project preset](https://pnp.github.io/pnpjs/concepts/project-preset/) as one method for creating and maintaining your factory interface object.

**Comment by Saintenr** [comment]
> Based on the little bit of code you shared I suspect that recreating the context every time a button is clicked is going to lead to issues, you might try reviewing this documentation on a [project preset](https://pnp.github.io/pnpjs/concepts/project-preset/) as one method for creating and maintaining your factory interface object.

Hi, i just found the issue and it seems to be my fault. 

i was trying to do this: 
`
// init search
const results = await sp.search({Querytext: 'test'}).getPage(1,10)

// and then later
results.getPage(2.10)
`
Apparently in v3 it is no longer possible to execute the getPage function on a getPage SearchResults. I have also tested this without context elsewhere and it really does not work. that was my mistake. As I said, in V2 it still worked. Maybe this was not intended?
Anyway, thanks for the tip that led me to the strange solution.


(I simplified the code to not include the context etc. however results was in my context as ref)
